src/textual/widget.py

```diff
@@ -363,6 +365,22 @@ def offset(self) -> Offset:
     def offset(self, offset: Offset) -> None:
         self.styles.offset = ScalarOffset.from_offset(offset)
 
+    def __enter__(self) -> None:
```

Comment:
Is there a reason why we can't return `self` from this? I'm thinking folk may want to sometimes hang on to a reference to the main wrapper widget for use elsewhere in the code:

```python
def compose(self) -> ComposeResult:
    with Horizontal(id="wrapper") as frame:
        self.frame = frame
        yield Foo()
        yield Bar()
        yield Baz()
```

Reply:
Good idea.